<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>trainDataWindow</class>
 <widget class="QDialog" name="trainDataWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>582</width>
    <height>411</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>380</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Train</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>380</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </widget>
  <widget class="QListWidget" name="batchList">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
  </widget>
  <widget class="QListWidget" name="expectedDataList">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>40</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>350</y>
     <width>551</width>
     <height>23</height>
    </rect>
   </property>
   <property name="value">
    <number>0</number>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>260</y>
     <width>119</width>
     <height>77</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Regularization:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="checkBox">
      <property name="text">
       <string>L1 </string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="checkBox_2">
      <property name="text">
       <string>L2</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="pushButton_3">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>250</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Add</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_4">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>250</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
  </widget>
  <widget class="QListWidget" name="inputDataList">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>40</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
  </widget>
  <widget class="QSpinBox" name="epochSpin">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>320</y>
     <width>101</width>
     <height>24</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>300</y>
     <width>57</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>Epochs:</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="learningRateSpin">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>270</y>
     <width>91</width>
     <height>24</height>
    </rect>
   </property>
   <property name="decimals">
    <number>7</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>250</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Learning Rate:</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="alphaSpin">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>270</y>
     <width>91</width>
     <height>24</height>
    </rect>
   </property>
   <property name="decimals">
    <number>7</number>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="betaSpin">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>320</y>
     <width>91</width>
     <height>24</height>
    </rect>
   </property>
   <property name="decimals">
    <number>7</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>250</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>L1 constant</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>300</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>L2 Constant</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Batch List: </string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>20</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Input Data: </string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>20</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Expected Data: </string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_3</sender>
   <signal>released()</signal>
   <receiver>trainDataWindow</receiver>
   <slot>addBatch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>released()</signal>
   <receiver>trainDataWindow</receiver>
   <slot>removeBatch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>167</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>171</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batchList</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>trainDataWindow</receiver>
   <slot>selectBatch(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>released()</signal>
   <receiver>trainDataWindow</receiver>
   <slot>trainNetwork()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>393</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>released()</signal>
   <receiver>trainDataWindow</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>397</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>375</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>clicked()</signal>
   <receiver>trainDataWindow</receiver>
   <slot>checked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>364</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>441</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_2</sender>
   <signal>clicked()</signal>
   <receiver>trainDataWindow</receiver>
   <slot>checked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>365</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>selectBatch(int)</slot>
  <slot>addBatch()</slot>
  <slot>removeBatch()</slot>
  <slot>editInputData(QString)</slot>
  <slot>editInputData(QListWidgetItem*)</slot>
  <slot>editExpectedData(QString)</slot>
  <slot>updateInputData(QListWidgetItem*,QListWidgetItem*)</slot>
  <slot>trainNetwork()</slot>
  <slot>checked()</slot>
 </slots>
</ui>
